core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorTableMetadata.java

```diff
@@ -29,6 +30,7 @@
     private final Optional<String> comment;
     private final List<ColumnMetadata> columns;
     private final Map<String, Object> properties;
+    private final List<String> checkConstraints;
```

Comment:
For enforcing constraints, having this on a table-level is enough, since "column constraints" (a constraint defined on a column) and "table constraints" (a constraint defined on one or multiple columns) do not need to be differentiated (provided their NULL behavior is same, it's non-obvious).

However, we will want to let connectors also create tables with constraints. Then, it may be needed to differentiate between them. I guess it would be nice to have both column-level and table-level constraints in ConnectorTableMetadata. 

(I am following PostgreSQL CREATE TABLE docs https://www.postgresql.org/docs/current/sql-createtable.html here.)

Reply:
Table check constraints and column check constraints are not equivalent. Column check constraints can only reference the column they are associated with.

The spec also talks about "enforced" vs "not enforced" constraints, so we may also need to be able to distinguish them. No sure what the semantics of a "not enforced" constraint are, though.

Reply:
> Table check constraints and column check constraints are not equivalent. Column check constraints can only reference the column they are associated with.

Thanks for pointing this out.
Let's do table constraints only for now and let's follow with column constraints later.

> The spec also talks about "enforced" vs "not enforced" constraints, so we may also need to be able to distinguish them. No sure what the semantics of a "not enforced" constraint are, though.

Let's support "enforced" constraints for now. 
Let's add support for "not enforced" later, when we need them.